apps/dashboard/app/views/batch_connect/session_contexts/_file_navigator.html.erb

```diff
@@ -1,4 +1,48 @@
+<%
+content_for :head do
+  javascript_include_tag 'navigator/index', nonce: true
+end
+%>
 
 <%= label attrib.id, "#{field_options[:label]}" %>
 <%= text_field :batch_connect_session_context, attrib.id, class: 'form-control' %>
-<%= button_tag 'Select Path', type: 'button', class: 'btn btn-primary', style: 'margin-top: 15px;', onclick: 'alert("Onclick Works Properly!");' %>
\ No newline at end of file
+<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#modelNavigator" id='select-nav-path'>
+  Select Path
+</button>
+<span id='navigate_home_dir' style="display:none;"><%= Dir.home %></span>
+<span id='navigate_url' style="display:none;"><%= ENV['RAILS_RELATIVE_URL_ROOT'] %></span>
+<hr>
```

Comment:
I would remove this, I'm not sure what purpose it serves, but it seems off.

```suggestion
```

Reply:
Thanks. Leftover from testing.